### ISSUE TITLE
Atttempting to access minishift from behind corporate proxy with self-signed cert

### ISSUE THREAD

**Issue by kylebrockmann** [minishift/minishift#1117]
Hi All,

I'm attempting to spin up a minishift off the Centos 7 Iso Image with the self-signed cert in the docker chain.

Error: error pulling Docker image openshift/origin:v1.5.1
   Caused By:
     Error: Error while pulling image: Get https://index.docker.io/v1/repositories/openshift/origin/images: x509: certificate signed by unknown authority

How can I get my root cert into the docker installation in the iso?

**Comment by hferentschik** [comment]
This is a duplicate of #391

**Comment by kylebrockmann** [comment]
For any of y'all who might be trapped behind a corporate proxy like myself, please find the procedure for installing trusted certs on minishift below. 
You can do this with a few shell scripts if you'd like as I did. 

minishift start --vm-driver=virtualbox 
minishift ssh "sudo curl <url_to_my_cert> -o /etc/pki/ca-trust/source/anchors/ca.crt"
minishift ssh "sudo update-ca-trust extract"
minishift ssh
sudo docker exec -i origin curl <url_to_my_cert> -o /etc/pki/ca-trust/source/anchors/ca.crt
sudo docker exec -i origin update-ca-trust extract
exit
minishift ssh "sudo systemctl restart docker"
minishift ssh "docker start origin"

**Comment by charrogruver** [comment]
@kylebrockmann Thanks for the tip!  However, It does not seem to work for me.  All of the steps up to "minishift stop" are successful.  However, on the second start, the VM seems to have not persisted any of the configuration.  

The second start of the VM throws: "Error starting the cluster: Error attempting to download and cache oc: Cannot get the OpenShift release version v1.5.1: Get https://api.github.com/repos/openshift/origin/releases/tags/v1.5.1: x509: certificate signed by unknown authority"

It appears that the VM did not save the fact that I installed the cert during the first set of steps.

The commands after "minishift ssh" fail with "Error response from daemon: No such container: origin"

Did you set any minishift config parameters before starting the VM for the first time?

**Comment by kylebrockmann** [comment]
@charrogruver The oc cert with gtihub also doesn't work. Try downloading the oc binary and putting it in your .minishift/cache/oc/<version> directory. That works for me as .minishift/cache/oc/v1.5.1/oc on linux or   .minishift/cache/oc/v1.5.1/oc.exe on windows.

**Comment by charrogruver** [comment]
@kylebrockmann 

That fixed it!  Thanks.  FYI: I'm using minishift 1.4.1 which is pulling origin v 3.6.0.  So, if and when you upgrade replace 1.5.1 with 3.6.0 in the above.

**Comment by kylebrockmann** [comment]
@charrogruver  Correct. I've done the same. v3.6.0 is quite gorgeous. Nice changes to the UI.
